Add deprecation notes in documentation (#2909)
jaegertracing/jaeger: CONTRIBUTING.md

Merge the PR by using "Squash and merge" option on Github. Avoid creating merge commits.
After the merge make sure referenced issues were closed.

> ## Deprecated CLI Flags

Author: This section is mostly a copy of the description from #2113.

> ## Deprecated CLI Flags

Reviewer: ```suggestion
## Deprecating CLI Flags
```

> * If a flag is deprecated in release N, it can be removed in release N+2.

Reviewer: ```suggestion
* If a flag is deprecated in release N, it can be removed in release N+2 or three months later, whichever is later.
```

> * When adding a (deprecated) prefix to the flags, indicate via a deprecation message that the flag could be removed after 3 months. For example:

Reviewer: ```suggestion
* When adding a (deprecated) prefix to the flags, indicate via a deprecation message that the flag could be removed in the future. For example:
```

  ```

>   deprecated, will be removed after 2020-03-15 or in release v1.19.0, whichever is later

Reviewer: ```suggestion
  (deprecated, will be removed after 2020-03-15 or in release v1.19.0, whichever is later)
```

  ```
* At the top of the file where the flag name is defined, add a constant and a comment, e.g.
  ```
  // TODO deprecated flag to be removed
  healthCheckHTTPPortWarning = "(deprecated, will be removed after 2020-03-15 or in release v1.19.0, whichever is later)"
  ```
* Use that constant as the prefix to the help text, e.g.
  ```
  flagSet.Int(healthCheckHTTPPort, 0, healthCheckHTTPPortWarning+" see --"+adminHTTPHostPort)
  ```

> * When reading the flag into config, log a warning with the same warning.

Reviewer: ```suggestion
* When parsing a deprecated flag into config, log a warning with the same deprecation message
```
